crypto/params.c

```diff
@@ -1217,41 +1245,42 @@ int OSSL_PARAM_set_double(OSSL_PARAM *p, double val)
     p->return_size = 0;
 
     if (p->data_type == OSSL_PARAM_REAL) {
-        p->return_size = sizeof(double);
-        if (p->data == NULL)
+        if (p->data == NULL) {
+            p->return_size = sizeof(double);
             return 1;
+        }
         switch (p->data_size) {
         case sizeof(double):
+            p->return_size = sizeof(double);
             *(double *)p->data = val;
             return 1;
         }
         err_unsupported_real;
         return 0;
     } else if (p->data_type == OSSL_PARAM_UNSIGNED_INTEGER) {
-        p->return_size = sizeof(double);
-        if (p->data == NULL)
+        if (p->data == NULL) {
+            /*
+             * Unclear how this is usable, the parameter's type is integral.
+             * Its size should be the size of some integral type.
+             */
+            p->return_size = sizeof(double);
             return 1;
+        }
         if (val != (uint64_t)val) {
             err_inexact;
             return 0;
         }
         switch (p->data_size) {
         case sizeof(uint32_t):
-            if (val >= 0 && val <= UINT32_MAX) {
+            if (val >= 0 && val < d_pow_32) {
```

Comment:
I much prefer the original comparison against `UINT32_MAX` here.  It's clearer IMO.

I get that you disagree...

Reply:
The comparison will be against the double value, may as well say so (cue G&S The Mikado).

Reply:
It increases cognative load.  UINT32_MAX is clear as to what is being tested.  The double gives no such indication, either in definition or use.